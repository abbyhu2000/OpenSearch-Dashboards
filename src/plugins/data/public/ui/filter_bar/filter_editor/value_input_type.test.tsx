/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { mountWithIntl } from 'test_utils/enzyme_helpers';
import { ValueInputType } from './value_input_type';

let onChangeMock: any;

describe('Value input type', () => {
  beforeAll(() => {
    onChangeMock = jest.fn();
  });
  it('is number', async () => {
    const valueInputProps = {
      value: 1,
      type: 'number',
      onChange: onChangeMock,
      onBlur: () => {},
      placeholder: '',
    };
    const component = mountWithIntl(<ValueInputType {...valueInputProps} />);
    expect(component).toMatchSnapshot();
  });

  it('is string', async () => {
    const valueInputProps = {
      value: 'value',
      type: 'string',
      onChange: () => {},
      onBlur: () => {},
      placeholder: '',
    };
    const component = mountWithIntl(<ValueInputType {...valueInputProps} />);
    expect(component).toMatchSnapshot();
  });

  it('is boolean', async () => {
    const valueInputProps = {
      value: 'true',
      type: 'boolean',
      onChange: () => {},
      onBlur: () => {},
      placeholder: '',
    };
    const component = mountWithIntl(<ValueInputType {...valueInputProps} />);
    expect(component).toMatchSnapshot();
  });
});
