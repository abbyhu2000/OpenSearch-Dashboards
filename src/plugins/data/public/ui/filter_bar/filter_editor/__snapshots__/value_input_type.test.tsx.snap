// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Value input type is boolean 1`] = `
<InjectIntl(ValueInputTypeUI)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
  onBlur={[Function]}
  onChange={[Function]}
  placeholder=""
  type="boolean"
  value="true"
>
  <ValueInputTypeUI
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {
          "date": Object {
            "full": Object {
              "day": "numeric",
              "month": "long",
              "weekday": "long",
              "year": "numeric",
            },
            "long": Object {
              "day": "numeric",
              "month": "long",
              "year": "numeric",
            },
            "medium": Object {
              "day": "numeric",
              "month": "short",
              "year": "numeric",
            },
            "short": Object {
              "day": "numeric",
              "month": "numeric",
              "year": "2-digit",
            },
          },
          "number": Object {
            "currency": Object {
              "style": "currency",
            },
            "percent": Object {
              "style": "percent",
            },
          },
          "relative": Object {
            "days": Object {
              "units": "day",
            },
            "hours": Object {
              "units": "hour",
            },
            "minutes": Object {
              "units": "minute",
            },
            "months": Object {
              "units": "month",
            },
            "seconds": Object {
              "units": "second",
            },
            "years": Object {
              "units": "year",
            },
          },
          "time": Object {
            "full": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "long": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "medium": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
            },
            "short": Object {
              "hour": "numeric",
              "minute": "numeric",
            },
          },
        },
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "onError": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": null,
      }
    }
    onBlur={[Function]}
    onChange={[Function]}
    placeholder=""
    type="boolean"
    value="true"
  >
    <EuiSelect
      onChange={[Function]}
      options={
        Array [
          Object {
            "text": "",
            "value": undefined,
          },
          Object {
            "text": "true",
            "value": "true",
          },
          Object {
            "text": "false",
            "value": "false",
          },
        ]
      }
      value="true"
    >
      <EuiFormControlLayout
        compressed={false}
        fullWidth={false}
        icon={
          Object {
            "side": "right",
            "type": "arrowDown",
          }
        }
        isLoading={false}
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <EuiValidatableControl>
              <select
                className="euiSelect"
                onChange={[Function]}
                onMouseUp={[Function]}
                value="true"
              >
                <option
                  key="0"
                />
                <option
                  key="1"
                  value="true"
                >
                  true
                </option>
                <option
                  key="2"
                  value="false"
                >
                  false
                </option>
              </select>
            </EuiValidatableControl>
            <EuiFormControlLayoutIcons
              compressed={false}
              icon={
                Object {
                  "side": "right",
                  "type": "arrowDown",
                }
              }
              isLoading={false}
            >
              <div
                className="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
              >
                <EuiFormControlLayoutCustomIcon
                  size="m"
                  type="arrowDown"
                >
                  <span
                    className="euiFormControlLayoutCustomIcon"
                  >
                    <EuiIcon
                      aria-hidden="true"
                      className="euiFormControlLayoutCustomIcon__icon"
                      size="m"
                      type="arrowDown"
                    >
                      <span
                        aria-hidden="true"
                        className="euiFormControlLayoutCustomIcon__icon"
                        data-euiicon-type="arrowDown"
                        size="m"
                      />
                    </EuiIcon>
                  </span>
                </EuiFormControlLayoutCustomIcon>
              </div>
            </EuiFormControlLayoutIcons>
          </div>
        </div>
      </EuiFormControlLayout>
    </EuiSelect>
  </ValueInputTypeUI>
</InjectIntl(ValueInputTypeUI)>
`;

exports[`Value input type is number 1`] = `
<InjectIntl(ValueInputTypeUI)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
  onBlur={[Function]}
  onChange={[MockFunction]}
  placeholder=""
  type="number"
  value={1}
>
  <ValueInputTypeUI
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {
          "date": Object {
            "full": Object {
              "day": "numeric",
              "month": "long",
              "weekday": "long",
              "year": "numeric",
            },
            "long": Object {
              "day": "numeric",
              "month": "long",
              "year": "numeric",
            },
            "medium": Object {
              "day": "numeric",
              "month": "short",
              "year": "numeric",
            },
            "short": Object {
              "day": "numeric",
              "month": "numeric",
              "year": "2-digit",
            },
          },
          "number": Object {
            "currency": Object {
              "style": "currency",
            },
            "percent": Object {
              "style": "percent",
            },
          },
          "relative": Object {
            "days": Object {
              "units": "day",
            },
            "hours": Object {
              "units": "hour",
            },
            "minutes": Object {
              "units": "minute",
            },
            "months": Object {
              "units": "month",
            },
            "seconds": Object {
              "units": "second",
            },
            "years": Object {
              "units": "year",
            },
          },
          "time": Object {
            "full": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "long": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "medium": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
            },
            "short": Object {
              "hour": "numeric",
              "minute": "numeric",
            },
          },
        },
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "onError": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": null,
      }
    }
    onBlur={[Function]}
    onChange={[MockFunction]}
    placeholder=""
    type="number"
    value={1}
  >
    <EuiFieldNumber
      onChange={[Function]}
      placeholder=""
      value={1}
    >
      <EuiFormControlLayout
        compressed={false}
        fullWidth={false}
        isLoading={false}
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <EuiValidatableControl>
              <input
                className="euiFieldNumber"
                onChange={[Function]}
                placeholder=""
                type="number"
                value={1}
              />
            </EuiValidatableControl>
            <EuiFormControlLayoutIcons
              compressed={false}
              isLoading={false}
            />
          </div>
        </div>
      </EuiFormControlLayout>
    </EuiFieldNumber>
  </ValueInputTypeUI>
</InjectIntl(ValueInputTypeUI)>
`;

exports[`Value input type is string 1`] = `
<InjectIntl(ValueInputTypeUI)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {
        "date": Object {
          "full": Object {
            "day": "numeric",
            "month": "long",
            "weekday": "long",
            "year": "numeric",
          },
          "long": Object {
            "day": "numeric",
            "month": "long",
            "year": "numeric",
          },
          "medium": Object {
            "day": "numeric",
            "month": "short",
            "year": "numeric",
          },
          "short": Object {
            "day": "numeric",
            "month": "numeric",
            "year": "2-digit",
          },
        },
        "number": Object {
          "currency": Object {
            "style": "currency",
          },
          "percent": Object {
            "style": "percent",
          },
        },
        "relative": Object {
          "days": Object {
            "units": "day",
          },
          "hours": Object {
            "units": "hour",
          },
          "minutes": Object {
            "units": "minute",
          },
          "months": Object {
            "units": "month",
          },
          "seconds": Object {
            "units": "second",
          },
          "years": Object {
            "units": "year",
          },
        },
        "time": Object {
          "full": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "long": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
            "timeZoneName": "short",
          },
          "medium": Object {
            "hour": "numeric",
            "minute": "numeric",
            "second": "numeric",
          },
          "short": Object {
            "hour": "numeric",
            "minute": "numeric",
          },
        },
      },
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "onError": [Function],
      "textComponent": Symbol(react.fragment),
      "timeZone": null,
    }
  }
  onBlur={[Function]}
  onChange={[Function]}
  placeholder=""
  type="string"
  value="value"
>
  <ValueInputTypeUI
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {
          "date": Object {
            "full": Object {
              "day": "numeric",
              "month": "long",
              "weekday": "long",
              "year": "numeric",
            },
            "long": Object {
              "day": "numeric",
              "month": "long",
              "year": "numeric",
            },
            "medium": Object {
              "day": "numeric",
              "month": "short",
              "year": "numeric",
            },
            "short": Object {
              "day": "numeric",
              "month": "numeric",
              "year": "2-digit",
            },
          },
          "number": Object {
            "currency": Object {
              "style": "currency",
            },
            "percent": Object {
              "style": "percent",
            },
          },
          "relative": Object {
            "days": Object {
              "units": "day",
            },
            "hours": Object {
              "units": "hour",
            },
            "minutes": Object {
              "units": "minute",
            },
            "months": Object {
              "units": "month",
            },
            "seconds": Object {
              "units": "second",
            },
            "years": Object {
              "units": "year",
            },
          },
          "time": Object {
            "full": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "long": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
              "timeZoneName": "short",
            },
            "medium": Object {
              "hour": "numeric",
              "minute": "numeric",
              "second": "numeric",
            },
            "short": Object {
              "hour": "numeric",
              "minute": "numeric",
            },
          },
        },
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {},
        "now": [Function],
        "onError": [Function],
        "textComponent": Symbol(react.fragment),
        "timeZone": null,
      }
    }
    onBlur={[Function]}
    onChange={[Function]}
    placeholder=""
    type="string"
    value="value"
  >
    <EuiFieldText
      onChange={[Function]}
      placeholder=""
      value="value"
    >
      <EuiFormControlLayout
        fullWidth={false}
      >
        <div
          className="euiFormControlLayout"
        >
          <div
            className="euiFormControlLayout__childrenWrapper"
          >
            <EuiValidatableControl>
              <input
                className="euiFieldText"
                onChange={[Function]}
                placeholder=""
                type="text"
                value="value"
              />
            </EuiValidatableControl>
            <EuiFormControlLayoutIcons />
          </div>
        </div>
      </EuiFormControlLayout>
    </EuiFieldText>
  </ValueInputTypeUI>
</InjectIntl(ValueInputTypeUI)>
`;
